django/middleware/csrf.py

```diff
@@ -154,10 +154,16 @@ def process_view(self, request, callback, callback_args, callback_kwargs):
                 if referer is None:
                     return self._reject(request, REASON_NO_REFERER)
 
+                # Here we generate a list of all acceptable HTTP referers,
+                # including the current host since that has been validated
+                # upstream.
+                #
                 # Note that request.get_host() includes the port.
-                good_referer = 'https://%s/' % request.get_host()
-                if not same_origin(referer, good_referer):
-                    reason = REASON_BAD_REFERER % (referer, good_referer)
+                good_hosts = list(settings.CSRF_TRUSTED_ORIGINS)
+                good_hosts.append(request.get_host())
+                good_referers = ['https://{0}/'.format(host) for host in good_hosts]
+                if not any([same_origin(referer, host) for host in good_referers]):
```

Comment:
Unnecessary list comprehension, `any(same_origin(referer, host) for host in good_referers)` will do.